[GlobalISel] Support vector G_UNMERGE_VALUES in computeKnownBits.

davemgreen opened the pull request:
This adds computeKnownBits support for vector->vector G_UNMERGE_VALUES, grabbing the known bits with an adjusted DemandedElts mask.

jayfoad reviewed (commented):
The implementation looks reasonable to me.

arsenm commented on llvm/lib/CodeGen/GlobalISel/GISelKnownBits.cpp:
```diff
@@ -514,23 +514,33 @@ void GISelKnownBits::computeKnownBitsImpl(Register R, KnownBits &Known,
-    KnownBits SrcOpKnown;
-    computeKnownBitsImpl(SrcReg, SrcOpKnown, DemandedElts, Depth + 1);
+    if (SrcTy.isVector() && SrcTy.getScalarType() != DstTy.getScalarType())
+      return; // TODO: Handle vector->subelement unmerges?
```
Yes, should

davemgreen committed:
[GlobalISel] Support vector G_UNMERGE_VALUES in computeKnownBits.

This adds computeKnownBits support for vector->vector G_UNMERGE_VALUES,
grabbing the known bits with an adjusted DemandedElts mask.
